doc/external-commands.rst

```diff
@@ -0,0 +1,9 @@
+External Commands
+=================
+
+Cabal provides a system for external commands, akin to the ones used by tools like ``git`` or ``cargo``.
+
+If you execute ``cabal my-custom-command`` before failing with the message ``Error: cabal: unrecognised command: my-custom-command (try --help)``, Cabal will search the path for an executable named ``cabal-my-custom-command``.
+If found, it will execute ``cabal exec cabal-my-custom-command``, passing any remaining arguments to this external command.
+
+For ideas or existing external commands, visit `this Discourse thread <https://discourse.haskell.org/t/an-external-command-system-for-cabal-what-would-you-do-with-it/7114>`_.
```

Comment:
The idea is good, maybe we need something less ephemeral than a discourse thread (a page on the cabal wiki? a simple document in the cabal repos? suggestions welcome)

Reply:
Maybe we can list them directly in `doc/external-commands.rst`?

Reply:
I would say no to a list in the manual because then we would have to maintain it and assure that the commands are in working condition :P 

[cargo](https://doc.rust-lang.org/cargo/reference/external-tools.html) does not list them in the manual.